.github/workflows/build-lint-test.yml

```
@@ -44,6 +44,17 @@ jobs:
       - name: 🧪 run import/requires tests
         run: npm run test:importRequire
 
+      - name: 📚 generate ABI docs
+        run: npx hardhat dodoc && npx ts-node scripts/interfaceIds.ts && npx prettier -w ./docs
```

can't you do just this here?

```suggestion
        run: npm run build:docs
```